core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreMainActivity.java

```diff
@@ -845,9 +845,11 @@ private void newTabInBackground(String url) {
   }
 
   private void closeTab(int index) {
-    currentWebViewIndex = 0;
     tempForUndo = webViewList.get(index);
     webViewList.remove(index);
+    if(currentWebViewIndex >= webViewList.size()){
```

Comment:
my thinking was
```
if (index <= currentWebviewIndex){
  currentWebViewIndex--;
}
```

did that not work?

Reply:
oh right, its removing a specific index. Missed that!

Reply:
hmm no what if the index is 0

Reply:
@Frans-Lukas did you see the above comment?

Reply:
oh, no I didnt. I'll fix!